### PR TITLE
Creation of Subscribers' API (WIP)

### PR DESCRIPTION
**Pull request by jclarkin**
- Work In Progress
- Attached Create, Read (one or list), and Delete to the router
- Responses need cleanup, and does not currently comply with proper JSON API format

### CODE REVIEW

**Comment by jclarkin** [comment]
Looking for feedback on first Golang code.

I know - Needs refactoring for test-ability.

File: src/notify/subscription/subscriptionDB.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+	_, err = prepStmts.destroy.Query(id)
+	return subscription, err
+}
+
+func prepareStatements() error {
+	var err error
+
+	prepStmts.insert, err = myDB.Prepare(`
+		INSERT INTO subscriptions
```
Comment (snocorp) [inline src/notify/subscription/subscriptionDB.go]: SQL in your code is fun.

File: src/notify/subscription/db.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+func CreateDB(db *sql.DB) error {
+	myDB = db
+
+	_, err := myDB.Exec(`
+              CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+              CREATE TABLE IF NOT EXISTS subscriptions (
+                id UUID PRIMARY KEY,
+                event_type VARCHAR(64),
+                context VARCHAR(64),
+                original_account_id VARCHAR(64),
```
Comment (asmaloney) [inline src/notify/subscription/db.go]: I believe this will be a UUID.  That's what I changed events to.

File: src/notify/subscription/db.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+	prepStmts.insert, err = myDB.Prepare(`
+		INSERT INTO subscriptions
+    ( id,
+      event_type,
+      context,
+      original_account_id
+    )
+    VALUES ( uuid_generate_v4(), $1, $2, $3 )
+  `)
```
Comment (asmaloney) [inline src/notify/subscription/db.go:111]: My formatting on these came out all wonky too...

File: src/notify/subscription/handlers.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	if err != nil {
+		jsonUtils.Error(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	subscriptionJSON, _ := json.Marshal(subscription)
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(201)
+	fmt.Fprintf(w, "%v", string(subscriptionJSON))
```
Comment (asmaloney) [inline src/notify/subscription/handlers.go]: These lines can be replaced with `jsonUtils.Output(w, 201, subscription)`.  See the event handlers.

(Same with the others below.)

File: src/notify/subscription/db.go (hunk 3 above)
Comment (jclarkin) [inline src/notify/subscription/db.go:111]: They look fine in my editor... just not on github.

image

Comment (asmaloney) [inline src/notify/subscription/db.go:111]: Yeah - me too.  Don't know why.

**Pull request merged**